python-client/giskard/datasets/base/__init__.py

```diff
@@ -204,13 +223,29 @@ def add_transformation_function(self, transformation_function: TransformationFun
         self.data_processor.add_step(transformation_function)
         return self
 
+    def dataset_hash(self):
+        if not self._dataset_hash_initialized:
```

Comment:
Do we need the `_dataset_hash_initialized`? Just checking if `self._hashes is None` would do, right?